### PR TITLE
initial commit of python test generator. closes #39, #40, #41, #42

### PR DESCRIPTION
**Pull request by sebastus**
The config file determines the tests to be run. The tests are codified in test_sub_*.py. main.py loads tests dynamically from pickled classes originating in test_sub_*.py files. Finally, event hub messages - generated from the required tests - are sent to the event hub specified in environment variables.

### CODE REVIEW

File: test-generator/generator-python/config.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (morsh) [inline test-generator/generator-python/config.json]: I think the folder structure should be `tests/test-generator/generator-python/config.json`

File: test-generator/generator-python/test_base_class.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'''
+This is the base class for all custom tests. 
+The generate method is abstract and must be supplied by any subclasses.
+'''
+from abc import ABC, abstractmethod
+
+class TestBaseClass(ABC):
```
Comment (morsh) [inline test-generator/generator-python/test_base_class.py]: Part of the classes are with Pascal case and some are with_underscore_casing, I think all classes should be Pascal case.

File: test-generator/generator-python/main.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import sys
```
Comment (morsh) [inline test-generator/generator-python/main.py]: I think there should be a comment explaining exactly what this model does, so devs won't have to search the methods for it

**Review by morsh — APPROVED** [review]
Just a few comments to fix.
Other than that: LGTM

**Review by morsh — CHANGES_REQUESTED** [review]
Changes requested

File: test-generator/generator-python/config.json (hunk 1 above)
Comment (sebastus) [reply test-generator/generator-python/config.json]: done.

File: test-generator/generator-python/test_base_class.py (hunk 2 above)
Comment (sebastus) [reply test-generator/generator-python/test_base_class.py]: I tried to follow advice as I was developing it - can we push this to another PR?

File: test-generator/generator-python/main.py (hunk 3 above)
Comment (sebastus) [reply test-generator/generator-python/main.py]: done

**Review by morsh — APPROVED** [review]
LGTM

**Pull request merged**